app/models/repository/destroying.rb

```diff
@@ -2,28 +2,34 @@ module Repository::Destroying
   extend ActiveSupport::Concern
 
   included do
-    @destroying ||= []
-    before_destroy :mark_as_destroying, prepend: true
-  end
-
-  def is_destroying?
-    self.class.instance_variable_get(:@destroying).include?(self.id)
+    scope :destroying, ->() { unscoped.where(is_destroying: true) }
+    scope :active, ->() { where(is_destroying: false) }
+    default_scope ->() { active }
   end
 
+  # Only use `destroy_asynchronously` if you want to destroy a repository.
+  # It prepares the deletion by setting a flag, which enables the deletion
+  # of its ontologies.
   def destroy
+    Rails.logger.info "Destroy #{self.class} #{self} (id: #{id})"
     super
   rescue StandardError => e
-    unmark_as_destroying
-    raise e.class, "Can't delete repository: It contains an #{Settings.OMS} that is imported by another repository."
+    self.is_destroying = false
+    save!
+    raise e.class, I18n.t('repository.delete_error', oms: Settings.OMS)
```

Comment:
Please instantiate a new error instance and set `e` as the origin so it won't get lost if you raise a new error